packages/support/src/Concerns/EvaluatesClosures.php

```diff
@@ -81,7 +82,10 @@ protected function resolveClosureDependencyForEvaluation(ReflectionParameter $pa
             return $this;
         }
 
-        if (filled($typedParameterClassName)) {
+        if (
+            filled($typedParameterClassName)
+            && ! is_subclass_of($typedParameterClassName, Model::class)
```

Comment:
If the model has been bound to the container manually, we should still resolve it. Someone might want to bind the current tenant to the container for easy access, for example.

Reply:
Good point, changed!